### PR TITLE
python rewrite of the simulator in order to better algin with other simulators

### PR DESCRIPTION
**Pull request by ianzhang366**
In order to achieve the following issue, we need to make sure the simulator will respond to a `ManagedCluster` Add an event.

https://github.com/open-cluster-management/backlog/issues/13877

Signed-off-by: Ian Zhang <izhang@redhat.com>

### CODE REVIEW

File: test/simulator/metrics-collector-simulator.py [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    api = hubClient.resources.get(api_version=groupVersion, kind=kind)
+
+    print("start watch: {}/{}".format(groupVersion, kind))
+
+    set = {}
+
+    for e in api.watch():
+        print(e['type'])
+
+        if e['type'] == "ADDED":
```
Comment (marcolan018) [inline test/simulator/metrics-collector-simulator.py:309]: do we need to watch "DELETED" event?

File: test/simulator/metrics-collector-simulator.py [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+                clusterRoleBindingAPI = hubClient.resources.get(
+                    api_version="rbac.authorization.k8s.io/v1",
+                    kind="ClusterRoleBinding")
+
+                clusterRoleBindingAPI.delete(
+                    name="{}-clusters-metrics-collector-view".format(
+                        clusterName))
+            except ApiException as err:
+                print("a", err.status)
+                if err.status != 404:
```
Comment (marcolan018) [inline test/simulator/metrics-collector-simulator.py]: if run into error, should we just log the error and continue to next cluster, but not return error?

File: test/simulator/metrics-collector-simulator.py (hunk 1 above)
Comment (ianzhang366) [reply test/simulator/metrics-collector-simulator.py:309]: I was thinking for using the `--clean` flag to delete the simulators, so I didn't watch the `DELETED`

File: test/simulator/metrics-collector-simulator.py (hunk 2 above)
Comment (ianzhang366) [reply test/simulator/metrics-collector-simulator.py]: yup, you are right, let it fall through makes more sense when deleting

**Comment by ianzhang366** [comment]
Signed-off-by: Ian Zhang <izhang@redhat.com>

**Comment by haoqing0110** [comment]
@ianzhang366 this PR still failed due to missing Signed-off-by. Should because some of your commits miss Signed-off-by. May need to squash all the 15 commits.

**Pull request closed without merging**